volume/volume_not_supported.go

```diff
@@ -89,3 +92,104 @@ func (s *quiesceNotSupported) Quiesce(
 func (s *quiesceNotSupported) Unquiesce(volumeID string) error {
 	return ErrNotSupported
 }
+
+type cloudBackupNotSupported struct{}
+
+// Backup uploads snapshot of a volume to cloud
+func (cl *cloudBackupNotSupported) Backup(
+	input *api.BackupRequest,
+) error {
+	return ErrNotSupported
+}
+
+// BackupRestore downloads a cloud backup to a newly created volume
+func (cl *cloudBackupNotSupported) BackupRestore(
+	input *api.BackupRestoreRequest,
+) *api.BackupRestoreResponse {
+	response := api.BackupRestoreResponse{
+		RestoreErr: ErrNotSupported.Error(),
+	}
+	return &response
+}
+
+// BackupEnumerate lists the backups for a given cluster/credential/volumeID
+func (cl *cloudBackupNotSupported) BackupEnumerate(
+	input *api.BackupEnumerateRequest,
+) *api.BackupEnumerateResponse {
+	response := api.BackupEnumerateResponse{
+		EnumerateErr: ErrNotSupported.Error(),
+	}
+	return &response
+}
+
+// BackupDelete deletes the backups in cloud
+func (cl *cloudBackupNotSupported) BackupDelete(
+	input *api.BackupDeleteRequest,
+) error {
+	return ErrNotSupported
+}
+
+// BackupStatus indicates the most recent status of backup/restores
+func (cl *cloudBackupNotSupported) BackupStatus(
+	input *api.BackupStsRequest,
+) *api.BackupStsResponse {
+	response := api.BackupStsResponse{
+		StsErr: ErrNotSupported.Error(),
+	}
+	return &response
+}
+
+// BackupCatalogue displays listing of backup content
+func (cl *cloudBackupNotSupported) BackupCatalogue(
+	cloudBackupID string,
+	credentialUUID string,
+) *api.BackupCatalogueResponse {
+	response := api.BackupCatalogueResponse{
+		CatalogueErr: ErrNotSupported.Error(),
+	}
+	return &response
+}
+
+// History displays past backup/restore operations in the cluster
+func (cl *cloudBackupNotSupported) BackupHistory(
+	volumeID string,
+) *api.BackupHistoryResponse {
+	response := api.BackupHistoryResponse{
+		HistoryErr: ErrNotSupported.Error(),
+	}
+	return &response
+}
+
+// CangeBackupState allows a current backup
+// state transisions(pause/resume/stop)
+func (cl *cloudBackupNotSupported) BackupStateChange(
+	volumeID string,
+	reqState string,
+) error {
+	return ErrNotSupported
+}
+
+// BackupSchedCreate for a volume creates a schedule to backup volume to cloud
+func (cl *cloudBackupNotSupported) BackupSchedCreate(
+	input *api.BackupSchedRequest,
+) *api.BackupSchedResponse {
+	response := api.BackupSchedResponse{
+		SchedCreateErr: ErrNotSupported.Error(),
+	}
+	return &response
+}
+
+// BackupSchedDelete delete a volume's cloud backup-schedule
```

Comment:
i think its not necessary to add comments here

Reply:
Ok